sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferStore.scala

```diff
@@ -40,7 +40,8 @@ object RapidsBufferStore {
  */
 abstract class RapidsBufferStore(
     val tier: StorageTier,
-    catalog: RapidsBufferCatalog = RapidsBufferCatalog.singleton)
+    catalog: RapidsBufferCatalog = RapidsBufferCatalog.singleton,
+    deviceStorage: RapidsDeviceMemoryStore = RapidsBufferCatalog.getDeviceStorage)
```

Comment:
Rather than tacking on the device store everywhere, would it be simpler to promote `getDeviceStorage` to be an overridable method on the catalog rather than a static method?

Reply:
Tried this, but there are a couple of places in shuffle manager that also calls `RapidsBufferCatalog.getDeviceStorage` directly, so we'd have to plumb the method through their catalogs. Not sure if it's an improvement.

Reply:
Cool, thanks for looking into it.  I agree that amount of change is probably not appropriate for this PR, and it's something we can address later.